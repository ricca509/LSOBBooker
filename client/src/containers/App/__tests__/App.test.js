import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { App } from '../';

describe('App container', () => {
  describe('when there is availability', () => {
    it('renders the data', () => {
      const wrapper = shallow(<App store={{
        availability: [{
          date: 'date',
          times: [1, 2]
        }],
        hasAvailability: true,
        location: 4,
        eventId: 68,
        fetching: false
      }} />);

      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('when there is no availability', () => {
    it('renders the loading spinner', () => {
      const wrapper = shallow(<App store={{
        availability: [],
        hasAvailability: false,
        location: 4,
        eventId: 68,
        fetching: false
      }} />);

      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
