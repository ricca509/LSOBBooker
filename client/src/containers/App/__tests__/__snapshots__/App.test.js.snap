exports[`App container when there is availability renders the data 1`] = `
<div
  className="container">
  <div
    className="row">
    <h3>
      London School of Barbering availability
    </h3>
  </div>
  <div
    className="row">
    <LocationHeader
      dates={
        Object {
          "end": "date",
          "start": "date",
        }
      }
      location="St Pauls/Farringdon - 6 West Smithfield EC1A 9JX" />
  </div>
  <div
    className="row">
    <Component
      availability={
        Array [
          Object {
            "date": "date",
            "times": Array [
              1,
              2,
            ],
          },
        ]
      }
      eventId={68}
      fetching={false}
      hasAvailability={true}
      location={4} />
  </div>
</div>
`;

exports[`App container when there is no availability renders the loading spinner 1`] = `
<div
  className="container">
  <div
    className="row">
    <h3>
      London School of Barbering availability
    </h3>
  </div>
  <div
    className="row">
    <LocationHeader
      dates={null}
      location="St Pauls/Farringdon - 6 West Smithfield EC1A 9JX" />
  </div>
  <div
    className="row">
    <Component />
  </div>
</div>
`;
